### ISSUE TITLE
Formatter doesn't wrap case statements in switch expressions

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#82090]
In current `master` (914948cbdea75eae44bc3564b828c849f7cc4437) I tried refactoring a switch statement into a statement that uses a switch expression instead, but the result of `./gradlew spotlessApply` leaves some excessively long lines:

```diff
diff --git a/test/framework/src/main/java/org/elasticsearch/cluster/coordination/AbstractCoordinatorTestCase.java b/test/framework/src/main/java/org/elasticsearch/cluster/coordination/AbstractCoordinatorTestCase.java
index db0c012980d..4bb580b567b 100644
--- a/test/framework/src/main/java/org/elasticsearch/cluster/coordination/AbstractCoordinatorTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/cluster/coordination/AbstractCoordinatorTestCase.java
@@ -1128,25 +1128,19 @@ public class AbstractCoordinatorTestCase extends ESTestCase {
                         TransportRequestOptions options,
                         DisruptableMockTransport destinationTransport
                     ) {
-                        final TransportRequestOptions.Type chanType = options.type();
-                        switch (action) {
-                            case JoinHelper.JOIN_ACTION_NAME:
-                            case FollowersChecker.FOLLOWER_CHECK_ACTION_NAME:
-                            case LeaderChecker.LEADER_CHECK_ACTION_NAME:
-                                assertThat(action, chanType, equalTo(TransportRequestOptions.Type.PING));
-                                break;
-                            case JoinHelper.JOIN_VALIDATE_ACTION_NAME:
-                            case PublicationTransportHandler.PUBLISH_STATE_ACTION_NAME:
-                            case PublicationTransportHandler.COMMIT_STATE_ACTION_NAME:
-                                assertThat(action, chanType, equalTo(TransportRequestOptions.Type.STATE));
-                                break;
-                            case JoinHelper.JOIN_PING_ACTION_NAME:
-                                assertThat(action, chanType, oneOf(TransportRequestOptions.Type.STATE, TransportRequestOptions.Type.PING));
-                                break;
-                            default:
-                                assertThat(action, chanType, equalTo(TransportRequestOptions.Type.REG));
-                                break;
-                        }
+                        assertThat(action, options.type(), switch (action) {
+                            case JoinHelper.JOIN_ACTION_NAME, FollowersChecker.FOLLOWER_CHECK_ACTION_NAME, LeaderChecker.LEADER_CHECK_ACTION_NAME -> equalTo(
+                                TransportRequestOptions.Type.PING
+                            );
+                            case JoinHelper.JOIN_VALIDATE_ACTION_NAME, PublicationTransportHandler.PUBLISH_STATE_ACTION_NAME, PublicationTransportHandler.COMMIT_STATE_ACTION_NAME -> equalTo(
+                                TransportRequestOptions.Type.STATE
+                            );
+                            case JoinHelper.JOIN_PING_ACTION_NAME -> oneOf(
+                                TransportRequestOptions.Type.STATE,
+                                TransportRequestOptions.Type.PING
+                            );
+                            default -> equalTo(TransportRequestOptions.Type.REG);
+                        });

                         super.onSendRequest(requestId, action, request, options, destinationTransport);
                     }
```

**Comment by pugnascotia** [comment]
As far as I can tell, there's no specific confirmation for this case, meaning that a change to the formatter itself would be necessary. I'd have to start with raising an Eclipse issue.

**Comment by pugnascotia** [comment]
Raised https://bugs.eclipse.org/bugs/show_bug.cgi?id=578044

**Comment by pugnascotia** [comment]
Fixed upstream in Eclipse, but not fixed in Spotless yet as of release 6.3.0.

**Comment by mark-vieira** [comment]
Resolved by #94994
